src/components/Image.ts

```diff
@@ -179,21 +298,105 @@ export class Image extends Component<ImageProps, ImageChild> {
 			);
 		}
 
-		const blipEmbedRel = evaluateXPathToString(
-			`${QNS.pic}blipFill/${QNS.a}blip/@${QNS.r}embed/string()`,
-			picNode,
-		);
-		const location = relationships.getTarget(blipEmbedRel);
-		const data = archive.readBinary(location);
+		const blipNode = evaluateXPathToFirstNode(`${QNS.pic}blipFill/${QNS.a}blip`, picNode);
+		if (blipNode === null) {
+			throw new Error('Failed to load image. No blip found inside a blipFill.');
+		}
+		const { main, svg } = extractData(archive, relationships, blipNode);
+
+		const dataExtensions: DataExtensions = {};
+		if (svg) {
+			dataExtensions.svg = svg.data;
+		}
+
 		const image = new Image({
-			data,
+			data: main.data,
+			dataExtensions,
 			title,
 			width,
 			height,
 		});
-		image.#location = location;
+		image.#embedMeta.location = main.location;
+		if (svg) {
+			const { svg: svgEmbed } = image.#embedMeta.extensions;
+			if (!svgEmbed) {
+				// At the time of writing this error should never happen.
+				// If you encountered it, it probably means that the Image constructor
+				// changed and is no longer defining #embed.svg when dataExtensions.svg
+				// is passed or this method changed and no longer passes
+				// dataExtensions.svg to the Image costuctor.
+				throw new Error(
+					'Failed setting image #embed.svg.location during Image deserialization. No SVG properties are available',
+				);
+			}
+			svgEmbed.location = svg.location;
+		}
 		return image;
 	}
 }
 
 registerComponent(Image as unknown as ComponentDefinition);
+
+type BlipLocationAndData = {
```

Comment:
Since `BlipLocationAndData` as well as `BlipSvgLocationAndData` are only ever referenced from `BlipAllLocationsAndData`, you might as well define them inline there. You can then also simplify the `BlipAllLocationsAndData` name somewhat maybe.